jscomp/ext/warnings.ml

```diff
@@ -490,8 +492,23 @@ let message = function
   | Bs_integer_literal_overflow ->
       "Integer literal exceeds the range of representable integers of type int"
   | Bs_uninterpreted_delimiters s -> "Uninterpreted delimiters " ^ s
-  | Bs_toplevel_expression_unit ->
-      "Toplevel expression is expected to have unit type."
+  | Bs_toplevel_expression_unit help ->
+      Printf.sprintf "This%sis at the top level and is expected to return `unit`. But, it's returning %s.\n\n  In ReScript, anything at the top level must evaluate to `unit`. You can fix this by assigning the expression to a value, or piping it into the `ignore` function.%s" 
```

Comment:
Sorry for reviewing so late 😞 and for nit-picking, but I think there should be no comma after "But".

Reply:
PRs welcome for late reviewers 😏

Reply:
Done in #6409